controllers/resource_controller.go

```diff
@@ -243,9 +243,12 @@ func (r *ResourceReconciler) reconcile(ctx context.Context, obj *v1alpha1.Resour
 		identity[k] = v
 	}
 
-	snapshotName, err := snapshot.GetSnapshotNameForObject(obj.GetName(), obj)
-	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("failed to generate snapshotname: %w", err)
+	snapshotName := obj.GetSnapshotName()
+	if snapshotName == "" {
```

Comment:
something is wrong if we get here and the snapshot name is empty, so I would just return an error here rather than generating the name at this point.

Reply:
Nevermind. :D